### PR TITLE
ENH: adding new function, `np.nanptp`

### PR DESCRIPTION
**Pull request by yliapis**
This PR adds an `nan` compatible analog function to `ptp`: `nanptp`

Do let me know if there are any areas for improvement.

### CODE REVIEW

File: numpy/lib/nanfunctions.py [hunk 1]
```diff
@@ -546,6 +548,73 @@ def nanargmax(a, axis=None):
...
+    When positive infinity and/or negative infinity are present:
+
+    >>> np.nanptp([1, 2, np.nan, np.inf])
+    inf
+    >>> np.nanptp([1, 2, np.nan, np.NINF])
+    inf
+
+    """
+    res = nanmax(a, axis=axis) - nanmin(a, axis=axis)
+    valid_mask = ~np.isnan(a)
```
Comment (mhvk) [inline numpy/lib/nanfunctions.py]: Is this necessary? Won't `nanmax` and `nanmin` already have each given a warning for all-nan axes?

Comment (yliapis) [reply numpy/lib/nanfunctions.py]: Looks like it isn't, I misunderstood the expected behavior in this case. I thought you need at least two values to have a valid peak-to-peak answer, but it looks like a single value should return `0`. I will fix this, thanks for catching it. Are the warnings from `nanmin` and `nanmax` acceptable, or do we want the function to have its own warning?

**Comment by shoyer** [comment]
I’m not sure this is worth the trouble. np.ptp is a relatively underused function, and is quite easy to write as a one liner (max minus min). If we were writing NumPy from scratch I probably wouldn’t bother to include the function at all given that we implement ptp in this exact same straightforward way (it might be different if we used some more efficient algorithm, e.g., that only needed a single pass over the data).

Are there cases where the correct behavior of nanptp would differ from `nanmax(a, axis) - nanmin(a, axis)`? I’m not convinced that we should issue warnings (beyond those already issued from nanmin/nanmax) or that it is correct to mask out results when there is only one valid value (that’s a perfectly well define peak to peak distance of zero).

**Comment by yliapis** [comment]
That is exactly how the function is written now, so it is pretty trivial to implement outside the library. The function could be twice as fast if we decided to dive to a lower level, though. The main argument I see for including the function is most functions performing reducing operations have a `nan*` analog.

**Comment by mhvk** [comment]
@yliapis - on the warnings, I guess there are two choices:
1. Rely on the `nanmin` and `nanmax` warnings for all-nan slices. Advantage: makes implementation trivial; Disadvantage: one gets two warnings.)
2. Copy the implementations over. Advantage: single warning. Disadvantage: additional complexity.

My sense would be to go with the simple route for now, but others may disagree...

**Comment by mattip** [comment]
Closing. @yliapis if you wish to pursue `nanptp` you should try to get support from the mailing list first, since three reviewers expressed reservations about including it.

**Pull request closed without merging**